src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -272,36 +272,34 @@ export class MockHttpSocket extends MockSocket {
     // if it hasn't been flushed already (e.g. someone started reading request stream).
     this.flushWriteBuffer()
 
-    const httpHeaders: Array<Buffer> = []
+    // Create a `ServerResponse` instance to delegate HTTP message parsing,
+    // Transfer-Encoding, and other things to Node.js internals.
+    const serverResponse = new ServerResponse(new IncomingMessage(this))
```

Comment:
I passed `this` socket instance as the socket of the `IncomingMessage` used here. I think it should be more error-proof in case this server response decides to terminate the socket (rightfully), for example.

Reply:
Ok. I used a new one because I was afraid of infinity loops, but if it makes sense to you I'm ok with this

Reply:
Should arrive at the infinite loops. This logic is only triggered on `respondWith()`, and that's called by the developer. If I spot any issues with this, will revert back to use an empty socket.

Reply:
Should arrive at the infinite loops. This logic is only triggered on `respondWith()`, and that's called by the developer. If I spot any issues with this, will revert back to use an empty socket.